ocaml/xapi/xapi_vm_group_helpers.ml

```diff
@@ -137,22 +137,25 @@ let update_vm_anti_affinity_alert_for_group ~__context ~group ~alerts =
       ()
 
 let maybe_update_vm_anti_affinity_alert_for_vm ~__context ~vm =
-  try
-    Db.VM.get_groups ~__context ~self:vm
-    |> List.filter (fun g ->
-           Db.VM_group.get_placement ~__context ~self:g = `anti_affinity
-       )
-    |> function
-    | [] ->
-        ()
-    | group :: _ ->
-        let alerts = get_anti_affinity_alerts ~__context in
-        let alerts_of_group =
-          filter_alerts_with_group ~__context ~group ~alerts
-        in
-        update_vm_anti_affinity_alert_for_group ~__context ~group
-          ~alerts:alerts_of_group
-  with e -> error "%s" (Printexc.to_string e)
+  if Pool_features.is_enabled ~__context Features.VM_group then
```

Comment:
The `maybe_update_vm_anti_affinity_alert_for_vm` is called when the feature status is changed:
When it is changed from `enabled` to `disabled`, `remove_vm_anti_affinity_alert`should be called. This assert will block this expected call.